program/rust/src/c_oracle_header.rs

```diff
@@ -34,94 +34,137 @@ pub trait PythAccount: Pod {
     }
 }
 
-impl PythAccount for pc_map_table_t {
+impl PythAccount for MappingAccount {
     const ACCOUNT_TYPE: u32 = PC_ACCTYPE_MAPPING;
     const INITIAL_SIZE: u32 = PC_MAP_TABLE_T_PROD_OFFSET as u32;
 }
 
-impl PythAccount for pc_prod_t {
+impl PythAccount for ProductAccount {
     const ACCOUNT_TYPE: u32 = PC_ACCTYPE_PRODUCT;
-    const INITIAL_SIZE: u32 = size_of::<pc_prod_t>() as u32;
+    const INITIAL_SIZE: u32 = size_of::<ProductAccount>() as u32;
     fn minimum_size() -> usize {
         PC_PROD_ACC_SIZE as usize
     }
 }
 
-impl PythAccount for pc_price_t {
+impl PythAccount for PriceAccount {
     const ACCOUNT_TYPE: u32 = PC_ACCTYPE_PRICE;
     const INITIAL_SIZE: u32 = PC_PRICE_T_COMP_OFFSET as u32;
 }
 
-#[cfg(target_endian = "little")]
-unsafe impl Zeroable for pc_acc {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Pod for pc_acc {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Zeroable for pc_map_table {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Pod for pc_map_table {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Zeroable for pc_prod {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Pod for pc_prod {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Zeroable for pc_price {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Pod for pc_price {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Zeroable for pc_price_info {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Pod for pc_price_info {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Zeroable for pc_ema {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Pod for pc_ema {
-}
-
-
-#[cfg(target_endian = "little")]
-unsafe impl Zeroable for pc_pub_key_t {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Pod for pc_pub_key_t {
-}
-
-
-#[cfg(target_endian = "little")]
-unsafe impl Zeroable for pc_price_comp_t {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Pod for pc_price_comp_t {
-}
-
-impl pc_pub_key_t {
-    pub fn new_unique() -> pc_pub_key_t {
+#[repr(C)]
+#[derive(Copy, Clone, Pod, Zeroable)]
+pub struct PriceAccount {
+    pub magic_:          u32,
+    pub ver_:            u32,
+    pub type_:           u32,
+    pub size_:           u32,
```

Comment:
Can we move these in-line comments above the fields and make them Rust doc comments:

```suggestion
    /// Example documentation comment with a triple ///
    pub size_:           u32,
```

Reply:
Good